packages/core/src/controllers/verify.ts

```diff
@@ -1,68 +1,140 @@
-    if (this.verifyUrl !== verifyUrl) {
-      this.removeIframe();
-    }
-    this.verifyUrl = verifyUrl;
+  get storeKey(): string {
+    return `verify:public:key`;
+  }
 
-    try {
-      await this.createIframe();
-    } catch (error) {
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+  public init = async () => {
+    this.publicKey = await this.store.getItem(this.storeKey);
+    console.log("persistedKey", this.publicKey);
+    if (this.publicKey && toMiliseconds(this.publicKey?.expiresAt) < Date.now()) {
+      console.log("public key expired");
+      await this.removePublicKey();
     }
+    if (this.publicKey) return;
+    const key = await this.fetchPublicKey();
+    console.log("public key", key);
+    await this.persistPublicKey(key);
   };
 
   public register: IVerify["register"] = async (params) => {
-    if (!this.initialized) {
-      this.addToQueue(params.attestationId);
-      await this.init();
-    } else {
-      this.sendPost(params.attestationId);
+    console.log("register", params);
+    const { id, decryptedId } = params;
+    const url = `${this.verifyUrlV2}/attestation?projectId=${this.projectId}`;
+    let src = "";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ id, decryptedId }),
+        headers: {
+          origin: "https://8951-78-130-198-143.ngrok-free.app/",
+        },
+      });
+      const { srcdoc } = await response.json();
+      src = srcdoc;
+    } catch (e) {
+      console.error("error", e);
+      return;
     }
+    console.log("srcdoc", src);
+
+    const abortTimeout = this.startAbortTimer(ONE_SECOND * 2);
+    const attestatiatonJwt = await new Promise((resolve) => {
+      const abortListener = () => {
+        document.body.removeChild(iframe);
+        window.removeEventListener("message", listener);
+        this.abortController.signal.removeEventListener("abort", abortListener);
+      };
+
+      this.abortController.signal.addEventListener("abort", abortListener);
+      const iframe = document.createElement("iframe");
+      iframe.srcdoc = src;
+      iframe.src = "https://verify.walletconnect.com";
+      iframe.style.display = "none";
+      const listener = (event: MessageEvent) => {
+        console.log("message event received", event);
+        if (!event.data) return;
+        const data = JSON.parse(event.data);
+        if (data.type === "verify_attestation") {
+          // best-practice field
+          clearInterval(abortTimeout);
+          window.removeEventListener("message", listener);
+          document.body.removeChild(iframe);
+          this.abortController.signal.removeEventListener("abort", abortListener);
+          console.log("attestation", data.attestation);
+          resolve(data.attestation === null ? "" : data.attestation);
```

Comment:
This mapping can be removed as per this conversation: https://walletconnect.slack.com/archives/C03TFK9BSGJ/p1722442807055809